### ISSUE TITLE
CPU thread support?

### ISSUE THREAD

**Issue by simonbyrne** [vchuravy/GPUifyLoops.jl#92]
Are we going to support the new CPU multithreading?

cc: @lcw

**Comment by vchuravy** [comment]
Eventually :)

**Comment by jpsamaroo** [comment]
@vchuravy you mentioned some issues to me about this during the hackathon. I recall that they were implementing shmem and scratch space. Can you elaborate a bit on these, and any others you can think about, for the benefit of the brave soul who chooses to take this one on?

**Comment by lcw** [comment]
One place to look is libocca, <https://libocca.org>, which uses a
similar kernel abstraction and has an OpenMP backend.

**Comment by vchuravy** [comment]
Yes shared memory and scratch memory are both abstractions that currently rely on the GPU hardware to do the right thing. Shared memory is shared across a block and scratch memory is thread local.

I think I have also seen codes in Clima that will require more care so that will take a review.
